### ISSUE TITLE
XML: representing "revision" in `<docidentifier>`

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-nist#73]
> SP 800-116r1 "Revision 1" is not indicated in the metadata (e.g., either in <docidentifier> or <docnumber>).
> I see <edition>1</edition> but it doesn't capture the designation of "Revision". 
> Also, I’m not sure how pubs that use "-n" as the edition/revision indicator would be reflected (e.g., SP 800-63-3).

Question:

1. Should the full document identifier be “NIST SP 800-161r1”? or “NIST SP 800-161 Revision 1”? In the case of SP 800-63-3, is it “NIST SP 800-63-3”? Does this mean the actual presentation of the document identifier can be arbitrary compared to the data model? If so we can make the full document identifier a free-form one.

2. It is true that the terms “edition” and “revision” are different. In ISO, a revised standard is however only indicated in an “edition” change. We have another concept of “iteration" in Metanorma as a “sub-edition” revision, such as the "second working draft" or “third draft standard”.

Thoughts?

**Comment by opoudjis** [comment]
The convention appears to be "docidentifier-partnumber Rev. revnumber" if there is a part number, and "docidentifier-revnumber" if there is not.

If there is a distinction to be made between edition and revision, I will need a formal definition. The use of the label "edition" vs "revision" on its own is not enough reason not to use the tag.

**Comment by ronaldtse** [comment]
I’m getting clarification but it seems that there are two representations of “revision” in a document ID - “SP 800-116 Revision 1” or “SP 800-63-3”. The first form seems to allow abbreviation to “SP 800-116 Rev. 1” or “SP 800-116r1”, but not for the second form. 

Maybe the full document ID and the abbreviated forms can be arbitrary and is not bound to a fixed representation (considering backwards compatibility here).

**Comment by opoudjis** [comment]
> Maybe the full document ID and the abbreviated forms can be arbitrary and is not bound to a fixed representation (considering backwards compatibility here).

All the more reason to impose a fixed representation going forward; but it could well be that we support a fully populated docidentifier attribute, and if it is absent, we use instead a docidentifier built out of the docnumber attributes and the other information about parts and revisions. But I need clear rules to do that. This should be an opportunity to improve consistency.

**Comment by ronaldtse** [comment]
Good idea. We'll wait for the user's confirmation to see if this is acceptable.

**Comment by ronaldtse** [comment]
The proposed solution would be to set a default:

* The user can set the document number and revision separately. By default, the document identifier will be `{document-number} Revision {revision-number}` and the abbreviation will be `{document-number} Rev. {revision-number}`.

* If the document uses an alternate scheme, the user can manually specify the full document identifier, such as `{document-number}-{revision-number}`, and also set to any abbreviation preferred, e.g. `{document-number}r{revision-number}`.

Solution acceptance confirmed ! 👍

**Comment by opoudjis** [comment]
This issue is now blocked pending confirmation.

**Comment by ronaldtse** [comment]
Yes.

**Comment by ronaldtse** [comment]
Clarification from NIST below:

* The full document identifier is spelled out, e.g., “NIST SP 800-161 Revision 1”.
* The use of the shorthand “NIST SP 800-161r1” is typically just used in the URL and DOI.
  * Some pubs like 800-63-3 have their revision number indicated as “-n”, so in this case the full document identifier is NIST SP 800-63-3.  I would not say that they’re arbitrary, necessarily, although our current usage not ideal. 
 
To accommodate both situations, does using the following approach make sense?

```xml 
<docidentifier type=”nist”>NIST SP 800-161 Revision 1</docidentifier>
<docnumber>SP 800-161</docnumber>  [Does having “SP” here make sense, or is it better to have in a separate field like <docseries>?]
<edition>Revision 1</edition>
``` 

[For the “-n” situations, keep the -n as part of the docnumber]
```xml 
<docidentifier type=”nist”>NIST SP 800-63-3</docidentifier>
<docnumber>SP 800-63-3</docnumber>
<edition></edition>
```
 
[And for situations involving “Parts”:]
```xml 
<docidentifier type=”nist”>NIST SP 800-57 Part 1 Revision 4</docidentifier>
<docnumber>SP 800-57 Part 1</docnumber>
<edition>Revision 4</edition>
```

The approaches above enable straightforward (hopefully) mapping from the <docnumber> and <edition> fields to the top two lines that appear on our Word/PDF cover and title pages, while the <docidentifier> (in combination with published-date and revdate) help us to uniquely identify the document.
 
This question is difficult for me to answer, since I don’t have a clear understanding of how the model is using all these different fields to produce the different outputs. I would be OK with the <docidentifier> field being freeform, for the time being.

**Comment by ronaldtse** [comment]
I propose we set a default:

* The user can set the document number and revision separately. By default, the document identifier will be `{document-number} Revision {revision-number}` and the abbreviation will be `{document-number} Rev. {revision-number}`.

* If the document uses an alternate scheme, the user can manually specify the full document identifier, such as `{document-number}-{revision-number}`, and also set to any abbreviation preferred, e.g. `{document-number}r{revision-number}`.

**Comment by ronaldtse** [comment]
The potential solution (https://github.com/metanorma/metanorma-nist/issues/73#issuecomment-477175279) is confirmed by NIST. Let's go with it!

(@opoudjis please provide comments on https://github.com/metanorma/metanorma-nist/issues/73#issuecomment-477167677)

**Comment by opoudjis** [comment]
My comment is that they are not allowing a general pattern for generating document identifiers, since they have exceptions. I agree that there should be a default pattern unless `:docidentifier:` is explicitly specified, and in fact I have already implemented that behaviour.

I am not generating both Revision and Rev; that isn't a requirement anywhere, since the DOI and URL are specified separately anyway. I am generating a long and a short version of the document identifier; the long version spells out NIST in full, and is used in the first line of output.

They guessed right: docnumber does not include SP, that goes to the `<series>`.

**Comment by ronaldtse** [comment]
@opoudjis I think there is some misunderstanding here.

There are two forms of the document identifier:

1. Full form:

Cover page:
<img width="561" alt="image" src="https://user-images.githubusercontent.com/11865/55449044-b0edba00-55fc-11e9-9fbc-105c3a6f53ec.png">

Title page (inner cover page):
<img width="556" alt="image" src="https://user-images.githubusercontent.com/11865/55449052-b519d780-55fc-11e9-9d07-301cbea92c65.png">

Authority:
<img width="518" alt="image" src="https://user-images.githubusercontent.com/11865/55449085-c9f66b00-55fc-11e9-85e4-bdf40ed62c40.png">

2. Abbreviated form:

Page headers:

<img width="550" alt="image" src="https://user-images.githubusercontent.com/11865/55449104-daa6e100-55fc-11e9-94b3-bd0ad45fe834.png">

**Comment by opoudjis** [comment]
This has been addressed in #88
